index.js

```diff
@@ -28,9 +29,25 @@ const supportedPreprocessors = [
   'sass'
 ];
 
+const componentDependencies = {
```

Comment:
Yes, somewhere is restricts the filter to just components and its templates. So when excluding everything, those things will still make it to the build, but checking if/which of these are needed based on the selection of whitelisted components would be a bit complicated, probably not worth the effort, and very brittle (we would need extensive tests to not suddenly break things)